docs/developer-guides/transactions/txscript/overview.md

```diff
@@ -0,0 +1,113 @@
+# <img class="dcr-icon" src="/img/dcr-icons/Code.svg" /> txscript
+
+---
+
+Decred uses a scripting system known as
+[txscript](https://github.com/decred/dcrd/tree/master/txscript) for
+transactions.
+txscript is a simple, stack-based,
+[Forth](https://en.wikipedia.org/wiki/Forth_(programming_language))-like
+programming language.
+The language is not Turing complete and does not implement any
+type of looping, but it is still fairly powerful despite this.
+Each of these properties are described in more detail below.
+
+A script can be considered as a list of conditions attached to a transaction,
+which must be met in order for the recipient of a transaction to spend the
+Decred they have received.
+
+In a simple transfer of Decred from one address to another, the script will
+typically specify just two requirements; in order to spend the received the
+Decred, the recipient must provide:
+
+- a public key that, when hashed, yields the receiving address specified in the
+  script
+- a signature to prove ownership of the private key corresponding to that public
+  key
+
+More complex scripts can be constructed requiring multiple private keys, or no
+keys at all, and it is possible to make use of time locks which prevent the
+spending of DCR until some point in the future.
+
+---
+
+## Stack-based
+
+txscript is described as a stack-based language because it uses a
+[stack](https://en.wikipedia.org/wiki/Stack_(abstract_data_type)) data structure
+in processing and executing commands.
+A given script is run from left-to-right, and as it runs it makes use of the
+stack to store values.
+
+txscript uses operation codes, or opcodes, to remove items from the stack,
+perform some calculation on them, and then push the result back onto the stack.
+
+To provide a simple example, consider the following script which uses the
+opcodes `OP_ADD` and `OP_EQUAL`.
+In the real world, the data used in scripts are usually cryptographic values
+such as public keys and signatures, but for the sake of simplicity this example
+just uses some small integers.
+
+```no-highlight
+6 5 OP_ADD 12 OP_EQUAL
+```
+
+This script contains five instructions, which will be executed as follows:
+
+1. The value 6 is pushed onto the stack
+  ![Simple addition step 1](/img/txscript/addition-1.png)
+
+1. The value 5 is pushed onto the stack
+  ![Simple addition step 2](/img/txscript/addition-2.png)
+
+1. `OP_ADD` pops the top two values from the stack, adds them together, and pushes
+   the result, 11, onto the stack
+  ![Simple addition step 3](/img/txscript/addition-3.png)
+
+1. The value 12 is pushed onto the stack
+  ![Simple addition step 4](/img/txscript/addition-4.png)
+
+1. `OP_EQUAL` pops the top two values from the stack, compares them, and pushes 0
+   (false) onto the stack. If the values were equal, 1 (true) would be pushed
+   instead.
+  ![Simple addition step 5](/img/txscript/addition-5.png)
```

Comment:
Awesome pics!

Don't know if you did them yourself or how hard it would be, but I suggest adding the script next to stack, with an indicator for the current opcode being executed and the sections that have executed and are still going to be executed.

![image](https://user-images.githubusercontent.com/1752493/92592458-fa244d80-f275-11ea-8c58-816b7ecf56c6.png)

Reply:
Nice idea, implemented. Preview site updated.

_(images are made the laziest way possible - LibreOffice Calc :stuck_out_tongue: )_